webapp/channels/src/components/sidebar/sidebar.test.tsx

```diff
@@ -113,4 +124,134 @@ describe('components/sidebar', () => {
 
         expect(wrapper).toMatchSnapshot();
     });
+
+    describe('unreads category', () => {
+        const currentUserId = 'current_user_id';
+
+        const channel1 = TestHelper.getChannelMock({id: 'channel1', team_id: currentTeamId});
+        const channel2 = TestHelper.getChannelMock({id: 'channel2', team_id: currentTeamId});
+
+        const baseState: DeepPartial<GlobalState> = {
+            entities: {
+                channels: {
+                    currentChannelId: channel1.id,
+                    channels: {
+                        channel1,
+                        channel2,
+                    },
+                    channelsInTeam: {
+                        [currentTeamId]: [channel1.id, channel2.id],
+                    },
+                    messageCounts: {
+                        channel1: {total: 10},
+                        channel2: {total: 10},
+                    },
+                    myMembers: {
+                        channel1: TestHelper.getChannelMembershipMock({channel_id: channel1.id, user_id: currentUserId, msg_count: 10}),
+                        channel2: TestHelper.getChannelMembershipMock({channel_id: channel2.id, user_id: currentUserId, msg_count: 10}),
+                    },
+                },
+                teams: {
+                    currentTeamId,
+                    teams: {
+                        [currentTeamId]: TestHelper.getTeamMock({id: currentTeamId}),
+                    },
+                },
+                users: {
+                    currentUserId,
+                },
+            },
+        };
+
+        test('should not render unreads category when disabled by user preference', () => {
+            const testState = mergeObjects(baseState, {
+                entities: {
+                    channels: {
+                        messageCounts: {
+                            [channel2.id]: {total: 15},
+                        },
+                    },
+                    preferences: {
+                        myPreferences: TestHelper.getPreferencesMock([
+                            {category: Preferences.CATEGORY_SIDEBAR_SETTINGS, name: Preferences.SHOW_UNREAD_SECTION, value: 'false'},
+                        ]),
+                    },
+                },
+            });
+
+            renderWithFullContext(
+                <Sidebar {...baseProps}/>,
+                mergeObjects(baseState, testState),
+            );
+
+            expect(screen.queryByText('UNREADS')).not.toBeInTheDocument();
+        });
+
+        test('should render unreads category when there are unread channels', () => {
+            const testState: DeepPartial<GlobalState> = {
```

Comment:
In the previous test, you merge the base state with this one here. In this text you do the merge when rendering. 0/5 on what is best, but for sure consistency is best :smile: